# ugr_ros_control Settings -----------------------
# Settings for ros_control control loop
generic_hw_control_loop:
  loop_hz: 300
  cycle_time_error_threshold: 0.01

# Settings for ros_control hardware interface
hardware_interface:
   drive_joint: axis_rear
   steering_joint: axis_steering
   joints:
      - axis_rear
      - axis_steering
   joints_config:
      axis_rear:
         interfaces:
            - velocity
            - effort
      axis_steering:
         interfaces:
            - position

# Publish all joint states ----------------------------------
# Creates the /joint_states topic necessary in ROS
joint_state_controller:
   type:         joint_state_controller/JointStateController
   publish_rate: 50

# Joint position controller for steering
steering_position_controller:
   type: position_controllers/JointPositionController
   joint: axis_steering

# controllers for autonomous driving
drive_velocity_controller:
   type: effort_controllers/JointVelocityController
   joint: axis_rear
   pid: {p: 1.0, i: 0.01, d: 0.0}

# controllers for manual driving
drive_effort_controller:
   type: effort_controllers/JointEffortController
   joint: axis_rear
